frontend/office-events.js

```diff
@@ -2,9 +2,15 @@ import io from "socket.io-client";
 
 function OfficeEvents(config) {
   this.config = config;
+
+  config.currentUser.imageUrl = encodeURIComponent(config.currentUser.imageUrl);
+
   const user = JSON.stringify(config.currentUser);
+
+  const queryConn = `user=${user}&room=${config.currentRoom}`;
+  
   this.socketIO = io.connect(config.domain, {
-    query: `user=${user}&room=${config.currentRoom}`
+    query: queryConn
```

Comment:
I think it can be solved using querystring (https://nodejs.org/api/querystring.html) node std lib.

```
 querystring.stringify({foo: '123', bar: 'hello', avatar: 'http://google.com/foo='})
'foo=123&bar=hello&avatar=http%3A%2F%2Fgoogle.com%2Ffoo%3D'
```
This way will encode url special chars, and don't need decode anything at frontend.

Reply:
I will fix in other PR with this suggestion...